core/src/main/scala/org/apache/spark/rpc/netty/NettyRpcEnv.scala

```diff
@@ -242,10 +244,10 @@ private[netty] class NettyRpcEnv(
         val rpcMessage = RpcOutboxMessage(message.serialize(this),
           onFailure,
           (client, response) => onSuccess(deserialize[Any](client, response)))
+        rpcMsg = Option(rpcMessage)
```

Comment:
what is this change for?

Reply:
To reference the `rpcMessage` within `onAbort()`:
```scala
def onAbort(t: Throwable): Unit = {
   onFailure(t)
   rpcMsg.foreach(_.onAbort())
}
```

Reply:
Why doesn't [this line](https://github.com/apache/spark/pull/28245/files#diff-0c89b4a60c30a7cd2224bb64d93da942L248) work?

Reply:
It works but this PR intends to remove `RpcAbortException` totally since it's really unnecessary.